UnitTests/MParticleWebViewTests.m

```diff
@@ -48,84 +36,59 @@ - (void)testInit {
+    [_webView startWithCustomUserAgent:@"Test User Agent" shouldCollect:NO defaultUserAgentOverride:nil];
     XCTAssertEqualObjects(_webView.userAgent, @"Test User Agent");
 }
 
 - (void)testUserAgentDisabled {
-    [_webView startWithCustomUserAgent:nil shouldCollect:NO defaultAgentOverride:nil];
+    [_webView startWithCustomUserAgent:nil shouldCollect:NO defaultUserAgentOverride:nil];
     NSString *defaultAgent = [NSString stringWithFormat:@"mParticle Apple SDK/%@", MParticle.sharedInstance.version];
     XCTAssertEqualObjects(_webView.userAgent, defaultAgent);
 }
 
 - (void)testUserAgentDefaultOverride {
-    [_webView startWithCustomUserAgent:nil shouldCollect:NO defaultAgentOverride:@"Test User Agent"];
+    [_webView startWithCustomUserAgent:nil shouldCollect:NO defaultUserAgentOverride:@"Test User Agent"];
     NSString *defaultAgent = [NSString stringWithFormat:@"mParticle Apple SDK/%@", MParticle.sharedInstance.version];
     XCTAssertNotEqualObjects(_webView.userAgent, defaultAgent);
     XCTAssertEqualObjects(_webView.userAgent, @"Test User Agent");
 }
 
-- (void)testUserAgentCapture {
-    MParticleWebView *mockWebView = OCMPartialMock(_webView);
-#if TARGET_OS_IOS == 1
-    [[(id)mockWebView expect] evaluateAgent];
-#else
-    [[(id)mockWebView reject] evaluateAgent];
-#endif
-    [mockWebView startWithCustomUserAgent:nil shouldCollect:YES defaultAgentOverride:nil];
-    [(id)mockWebView verify];
-}
-
 - (void)testShouldCollectResolved {
-    _webView.resolvedAgent = @"Test User Agent";
+    _webView.resolvedUserAgent = @"Test User Agent";
     XCTAssertFalse([_webView shouldDelayUpload:5]);
 }
 
 - (void)testShouldCollectPending {
-    _webView.resolvedAgent = nil;
+    _webView.resolvedUserAgent = nil;
     _webView.isCollecting = YES;
     _webView.initializedDate = [NSDate date];
     XCTAssertTrue([_webView shouldDelayUpload:5]);
 }
 
 - (void)testShouldCollectNoDate {
-    _webView.resolvedAgent = nil;
+    _webView.resolvedUserAgent = nil;
     _webView.isCollecting = YES;
     _webView.initializedDate = nil;
     XCTAssertFalse([_webView shouldDelayUpload:5]);
 }
 
 - (void)testShouldCollectTooLong {
-    _webView.resolvedAgent = nil;
+    _webView.resolvedUserAgent = nil;
     _webView.isCollecting = YES;
     _webView.initializedDate = [NSDate dateWithTimeIntervalSinceNow:-6];
     XCTAssertFalse([_webView shouldDelayUpload:5]);
 }
 
 - (void)testShouldCollectTimeLeft {
-    _webView.resolvedAgent = nil;
+    _webView.resolvedUserAgent = nil;
     _webView.isCollecting = YES;
     _webView.initializedDate = [NSDate dateWithTimeIntervalSinceNow:-4];
     XCTAssertTrue([_webView shouldDelayUpload:5]);
 }
 
 - (void)testOriginalDefaultAgent {
     NSString *defaultAgent = [NSString stringWithFormat:@"mParticle Apple SDK/%@", MParticle.sharedInstance.version];
-    XCTAssertEqualObjects(_webView.originalDefaultAgent, defaultAgent);
-}
-
-- (void)testBackgroundCollection {
```

Comment:
Yesh, so the 2 tests I removed were because OCMock wasn't properly detecting the call of the method (the "expect") even though it was 100% being called, confirmed via breakpoints. Seems to be some incompatibility with OCMock and Swift. In any case, when I looked closer at what was really being tested, it was just testing that a method was called that is literally always called, it wasn't really testing the logic.

Basically on iOS `evaluateAgent` is always called after calling `startWithCustomUserAgent` and on tvOS it never is, so this and the other test I removed ultimately were just testing that the `#if os(iOS)` worked, which isn't really useful.